generic3g/MAPL_Generic.F90

```diff
@@ -574,26 +586,271 @@ subroutine gridcomp_connect_all(gridcomp, src_comp, dst_comp, rc)
       _RETURN(_SUCCESS)
    end subroutine gridcomp_connect_all
 
-   subroutine hconfig_get_string(hconfig, keystring, value, default, rc)
-      type(ESMF_HConfig), intent(inout) :: hconfig
-      character(*), intent(in) :: keystring
-      character(:), allocatable :: value
-      character(*), optional, intent(in) :: default
+   subroutine gridcomp_get_hconfig(gridcomp, hconfig, rc)
+      type(ESMF_GridComp), intent(inout) :: gridcomp
+      type(ESMF_HConfig), intent(out) :: hconfig
       integer, optional, intent(out) :: rc
 
       integer :: status
-      logical :: has_key
+      type(ESMF_Config) :: config
       
-      has_key = ESMF_HConfigIsDefined(hconfig, keystring=keystring, _RC)
-      if (has_key) then
-         value = ESMF_HConfigAsSTring(hconfig, keystring=keystring, _RC)
-         _RETURN(_SUCCESS)
-      end if
+      call ESMF_GridCompGet(gridcomp, config=config, _RC)
+      call ESMF_ConfigGet(config, hconfig=hconfig, _RC)
+
+      _RETURN(_SUCCESS)
+   end subroutine gridcomp_get_hconfig
+
+   subroutine mapl_resource_get_i4_gc(gc, keystring, value, unusable, default, value_set, rc)
+      integer(kind=ESMF_KIND_I4), intent(inout) :: value
+      integer(kind=ESMF_KIND_I4), optional, intent(in) :: default
+      type(ESMF_GridComp), intent(inout) :: gc
+      character(len=*), intent(in) :: keystring
+      class(KeywordEnforcer), optional, intent(in) :: unusable
+      logical, optional, intent(out) :: value_set
+      integer, optional, intent(out) :: rc
+      class(Logger_t), pointer :: logger
+      type(ESMF_HConfig) :: hconfig
+      type(HConfigParams) :: params
+      integer :: status
+
+      call MAPL_GridCompGet(gc, hconfig=hconfig, logger=logger, _RC)
+      params = HConfigParams(hconfig, keystring, value_set, logger)
+      call MAPL_HConfigGet(params, value, default, _RC) 
+      if(present(value_set)) value_set = params%value_set
+
+      _RETURN(_SUCCESS)
+      _UNUSED_DUMMY(unusable)
+
+   end subroutine mapl_resource_get_i4_gc
+
+   subroutine mapl_resource_get_i8_gc(gc, keystring, value, unusable, default, value_set, rc)
+      integer(kind=ESMF_KIND_I8), intent(inout) :: value
+      integer(kind=ESMF_KIND_I8), optional, intent(in) :: default
+      type(ESMF_GridComp), intent(inout) :: gc
+      character(len=*), intent(in) :: keystring
+      class(KeywordEnforcer), optional, intent(in) :: unusable
+      logical, optional, intent(out) :: value_set
+      integer, optional, intent(out) :: rc
+      class(Logger_t), pointer :: logger
+      type(ESMF_HConfig) :: hconfig
+      type(HConfigParams) :: params
+      integer :: status
+
+      call MAPL_GridCompGet(gc, hconfig=hconfig, logger=logger, _RC)
+      params = HConfigParams(hconfig, keystring, value_set, logger)
+      call MAPL_HConfigGet(params, value, default, _RC) 
```

Comment:
This is not what we discussed the other day.   HConfigGet should take an hconfig object.

Reply:
Will change in morning.

Reply:
Working on implementing change.